### PR TITLE
Fix keybinds to special characters not showing in keybinding panel

### PR DESCRIPTION
**Pull request by WeirdRubberDuck**

### CODE REVIEW

File: src/components/BottomBar/KeybindingPanel.jsx [hunk 1]
```diff
@@ -129,16 +121,15 @@ class KeybindingPanel extends Component {
...
       return key;
     } else {
       return "{" + key.toLowerCase() + "}";
     }
-    return "";
   }
 
   specialKeyMatch = (key, actionKey) => {
-   var strippedKey = key.substr(1,key.indexOf('}') -1);
+    let strippedKey = key.substr(1,key.indexOf('}') -1);
```
Comment (ylvaselling) [inline src/components/BottomBar/KeybindingPanel.jsx:132]: `key.substr(1, key.indexOf('}') - 1);`

File: src/components/BottomBar/KeybindingPanel.jsx [hunk 2]
```diff
@@ -154,39 +145,36 @@ class KeybindingPanel extends Component {
...
+      shift: this.state.activeModifiers.includes('shift'),
+      super: this.state.activeModifiers.includes('super')
     }
-    return false;
+    const hasModifier = (action.modifiers["super"] || action.modifiers["alt"] || 
+                         action.modifiers["control"] || action.modifiers["shift"]);
+
+    const showForModifiers = Object.entries(modifierObject).toString() === Object.entries(action.modifiers).toString();
+
+    return showForModifiers || (!hasModifier && this.state.activeModifiers.length == 0);
```
Comment (ylvaselling) [inline src/components/BottomBar/KeybindingPanel.jsx:159]: Make the second part of the statement into bool? A little hard to understand what it means currently

File: src/components/BottomBar/KeybindingPanel.jsx [hunk 3]
```diff
@@ -154,39 +145,36 @@ class KeybindingPanel extends Component {
...
             keyActions.push(action);
           }
         }
       } 
     }
 
-    var actionActions = [];
-    for (var keyAction of keyActions) {
-      var matched = this.props.actions.data.shortcuts.filter((action) => {
+    let actionActions = [];
```
Comment (ylvaselling) [inline src/components/BottomBar/KeybindingPanel.jsx:175]: What does actionAction mean?

**Comment by Roxeena** [comment]
👍

**Review by alexanderbock — APPROVED** [review]
Works for me!

File: src/components/BottomBar/KeybindingPanel.jsx (hunk 3 above)
Comment (WeirdRubberDuck) [reply src/components/BottomBar/KeybindingPanel.jsx]: Haha, very good question... It seems to get the actual information about the action, while the loop above contains the keybinding information

This code is a bit mysterious to me (Micah wrote it). Added some comments and just renamed the array to "actions"

**Pull request merged**